woocommerce/woocommerce-gutenberg-products-block
# Featured Product Block: Make it possible to feature a specific product variation

What is done:

- Search component returns variations and shows counts where applicable
- Product search toggles variation display on row click (cleared this part with Jay)
- Variations display correctly front and back, inheriting the parent image if no variation image is set
- Variations show the correct description
- Variation cart buttons link to single product page with options pre-selected

Had some API issues but I worked around them <ins>by editing the blocks endpoints</ins>.

- [x] <del>This currently relies on v4 endpoints in order to get data about variations in bulk. This should probably move to the blocks API</del>
- [x] <del>There is currently no way to get parent ID or parent images so variations can inherit parent product images if they have none of their own</del>
- [x] <del>Short description/description is not consistent for variations. I have it working but I think it should be tackled in the API instead.</del>

I ended up using `MenuItem` directly instead of SearchListItem to support the display of the variation counts.


Closes #369

## Screenshots

![2019-07-02 14 19 30](https://user-images.githubusercontent.com/90977/60515959-a06cde80-9cd4-11e9-826e-9a0090ebd9b8.gif)

## How to test

Add a featured product block (repeat multiple times to check the following scenarios):

1. Select a simple product. Click done. Check display.
2. Select a variable product. Ensure the variations display below it. Click done. Check display.
3. Select a variable product. Then select a variation. Click done. Check display.
